sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/DoNotCatchNullReferenceException.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        int Foo(string s)
```

Comment:
We may want to also support exception filters, e.g: 
`catch (Exception e) when (e is NullReferenceException)`
But it seems a very unlikely case, so if it's not trivial, it's fine to skip it.

But we need test case documenting the behaviour.